pkg/local/sync.go

```diff
@@ -304,18 +305,35 @@ func (s *SyncManager) upload(ctx context.Context, rootPath string, remote *uri.U
 	metadata := map[string]string{
 		ClientMtimeMetadataKey: strconv.FormatInt(fileStat.ModTime().Unix(), 10),
 	}
-	reader := fileWrapper{
+
+	reader := b.Reader(f)
+	if s.includePerm {
+		if strings.HasSuffix(path, uri.PathSeparator) { // Create a 0 byte reader for directories
```

Comment:
Why not to create an empty reader for dirs even when `includePerm == false` - 
Mainly to prevent future bugs?

Reply:
1. Yes
2. I want to isolate the code that we'll need to remove later as much as possible